Lib/test/test_importlib/resources/test_files.py

```diff
@@ -79,14 +79,14 @@ def test_module_resources(self):
         A module can have resources found adjacent to the module.
         """
         spec = {
-            'mod.py': '',
+            'mod_res.py': '',
             'res.txt': 'resources are the best',
         }
         _path.build(spec, self.site_dir)
-        import mod
+        import mod_res
 
-        actual = resources.files(mod).joinpath('res.txt').read_text(encoding='utf-8')
```

Comment:
How is the name of this module relevant to the reported issue? My preference is to keep the name short and sweet.

Reply:
I see now db458522b503a78a0faa5b5c9d26b5cd9c27b627 indicates this is a workaround for importlib tests failing. That's interesting and surprising, because I'd expect `self.site_dir` to be unique for each test. My guess is it's getting masked by some other test. It's probably that other test that should be amended not to leak state.

Reply:
The interaction of these two tests doesn't affect the upstream implementations, as the two implicated tests are in different packages (`test_resolve` comes from `importlib_metadata` and `test_module_resources` comes from `importlib_resources`).

Reply:
Digging deeper into what's going on here, I see that both tests are creating a `mod` module on `sys.path` ([here](https://github.com/python/cpython/blob/c012c8ab7bb72a733bd98be5df32e262b9045f1a/Lib/test/test_importlib/fixtures.py#L98-L123) and [here](https://github.com/python/cpython/blob/c012c8ab7bb72a733bd98be5df32e262b9045f1a/Lib/test/test_importlib/resources/test_files.py#L67-L73)).

And it seems it's the latter that's not cleaning itself up properly, because it's the presence of that empty `mod` that's interfering with the entry point loading.

Reply:
I guess I'd expected the `CleanImport` context to provide some isolation for the test, but reviewing [the implementation](https://github.com/python/cpython/blob/c012c8ab7bb72a733bd98be5df32e262b9045f1a/Lib/test/support/import_helper.py#L186), it only _updates_ `sys.modules` but doesn't clean out any new imports made in the context. :(

Reply:
Thanks for chiming in, Jason!

> I see now https://github.com/python/cpython/commit/db458522b503a78a0faa5b5c9d26b5cd9c27b627 indicates this is a workaround for importlib tests failing. That's interesting and surprising, because I'd expect self.site_dir to be unique for each test. My guess is it's getting masked by some other test.

Yes, the referenced commit is a _workaround_, as I already stated in https://github.com/python/cpython/pull/116307#issuecomment-1979792609.

> It's probably that other test that should be amended not to leak state.

I don't care which one gets amended. But IMO, we should amend the cleanup fixtures, not the tests.

> And it seems it's the latter that's not cleaning itself up properly, because it's the presence of that empty mod that's interfering with the entry point loading.

Correct.

> I guess I'd expected the CleanImport context to provide some isolation for the test, but reviewing [the implementation](https://github.com/python/cpython/blob/c012c8ab7bb72a733bd98be5df32e262b9045f1a/Lib/test/support/import_helper.py#L186), it only updates sys.modules but doesn't clean out any new imports made in the context. :(

I also expected this. I hacked around with the fixtures yesterday, but I could not get them to clean up properly, hence I pushed the hack/workaround.

Reply:
In eerlend-aasland/cpython#61, I've proposed a change to this PR to add a new fixture and use it instead of `CleanImport`.

Reply:
Thanks; merged!

Reply:
I've reverted my workaround from yesterday in 39d135d15a14c47e34d254c325eaecb7c6214445. Thanks for helping, Jason. Highly appreciated.